File: video/ai-extensions/video-59-namegen/namegen.swift
import SwiftUI

struct ContentView: View {
  @State private var backgroundUrl: URL?
  @State private var names: [String] = []
  @State private var inputText: String = ""

  var body: some View {
    VStack {
      if let backgroundUrl = backgroundUrl {
        // AsyncImage is a SwiftUI view that can load and display an image from a URL.
        // It's new in iOS 15. If you're targeting earlier iOS versions, you'll need to handle image loading differently.
        AsyncImage(url: backgroundUrl) { image in
          image.resizable()
        } placeholder: {
          Color.gray
        }
          .frame(width: 300, height: 300)
          .cornerRadius(10)
        }

        TextField("Enter idea", text: $inputText)
          .textFieldStyle(RoundedBorderTextFieldStyle())
          .padding()

        Button(action: generateNames) {
          Text("Generate")
            .foregroundColor(.white)
            .padding()
            .background(Color.blue)
            .cornerRadius(10)
        }

        List(names, id: .self) { name in
          Text(name)
        }
      }
      .onAppear(perform: loadBackground)
  }

  func loadBackground() {
    // Here you would call the Unsplash API to get a random image URL
    // and update the backgroundUrl state property.
    // Replace "YOUR_UNSPLASH_ACCESS_KEY" with your actual Unsplash API key.
    let urlString = "https://api.unsplash.com/photos/random?client_id=YOUR_UNSPLASH_ACCESS_KEY&query=inspiration"
    guard let url = URL(string: urlString) else { return }

    // Omitting the network call for simplicity. You would parse the JSON response
    // and use it to set the backgroundUrl property.
  }

  func generateNames() {
    names = ["Innova", "Creativio", "IdeaStorm", "Conceptualize", "Brainwave"]
  }
}

struct ContentView_Previews: PreviewProvider {
  static var previews: some View {
    ContentView()
  }
}

File: video/ai-extensions/video-59-namegen/namegen.js
import { useState, useEffect } from "react";
import Image from "next/image";
import styles from "../styles/Home.module.css"; // or your custom CSS module

export default function Home() {
  const [background, setBackground] = useState("");
  const [names, setNames] = useState([]);

  // Function to fetch images from Unsplash and update the background
  useEffect(() => {
    async function updateBackground() {
      const response = await fetch(
        "https://api.unsplash.com/photos/random?client_id=YOUR_UNSPLASH_ACCESS_KEY&query=inspiration"
      );
      const data = await response.json();
      setBackground(data.urls.regular);
    }

    updateBackground();
  }, []);

  // Function to generate names (placeholder function)
  function generateNames() {
    const ideas = [
      "Innova",
      "Creativio",
      "IdeaStorm",
      "Conceptualize",
      "Brainwave",
    ];
    setNames(ideas);
  }

  return (
    <div
      className={styles.container}
      style={{ backgroundImage: `url(${background})` }}
    >
      <div className={styles.card}>
        <h1 className={styles.title}>Name Generator App</h1>
        <input
          type="text"
          id="ideaInput"
          placeholder="Enter idea"
          className={styles.input}
        />
        <button onClick={generateNames} className={styles.button}>
          Generate
        </button>
        <ul className={styles.nameList}>
          {names.map((name, index) => (
            <li key={index} className={styles.nameItem}>
              {name}
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
}
